packages/wonder-blocks-core/util/unique-id-factory.js

```diff
@@ -0,0 +1,58 @@
+// @flow
+import {IIdentifierFactory} from "../util/types.js";
+
+/**
+ * This is NOT for direct use. Instead, see the UniqueIDProvider component.
+ *
+ * Implements IIdentifierFactory to provide unique identifiers.
+ */
+export default class UniqueIDFactory implements IIdentifierFactory {
+    _uniqueFactoryName: string;
+
+    static _factoryUniquenessCounter = 0;
+
+    /**
+     * Creates a UniqueIDFactory instance.
+     *
+     * @param {string} factoryName An optional case-insensitive name for the
+     * factory. This will be used as part of the identifier. Useful for
+     * providing context to the identifiers, which can be useful in
+     * differentiating elements when debugging the DOM. This must contain only
+     * hyphen and alphanumeric characters.
+     */
+    constructor(scope?: string) {
+        scope = typeof scope === "string" ? scope : "";
+        const normalizedScope = scope.toLowerCase();
+        if (!this._hasValidIdChars(normalizedScope)) {
+            throw new Error(`Invalid factory name: ${scope}`);
+        }
+        this._uniqueFactoryName = `uid-${normalizedScope}-${UniqueIDFactory._factoryUniquenessCounter++}`;
+    }
+
+    _hasValidIdChars(value: ?string) {
```

Comment:
Maybe `_isValidId`?

Reply:
This was copied directly from our webapp implementation. I didn't want to mislead that we were checking anything but the character make-up of the ID. `_isValidId` implies it might check more about what makes an identifier valid (such as enforcing no leading numbers).